daemon/src/engine/framework/Rcon.cpp

```diff
@@ -0,0 +1,252 @@
+Daemon BSD Source Code
+Copyright (c) 2013-2016, Daemon Developers
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions are met:
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of the Daemon developers nor the
+      names of its contributors may be used to endorse or promote products
+      derived from this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL DAEMON DEVELOPERS BE LIABLE FOR ANY
+DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+===========================================================================
+*/
+#include "common/Common.h"
+#include "engine/qcommon/qcommon.h"
+#include "Rcon.h"
+#include "Crypto.h"
+#include "framework/Network.h"
+#include "engine/server/CryptoChallenge.h"
+
+namespace Rcon {
+
+Cvar::Cvar<std::string> cvar_server_password(
+    "rcon.server.password",
+    "Password used to protect the remote console",
+    Cvar::NONE,
+    ""
+);
+
+Cvar::Range<Cvar::Cvar<int>> cvar_server_secure(
+    "rcon.server.secure",
+    "How secure the Rcon protocol should be: "
+        "0: Allow unencrypted rcon, "
+        "1: Require encryption, "
+        "2: Require encryption and challege check",
+    Cvar::NONE,
+    0,
+    0,
+    2
+);
+
+Cvar::Cvar<std::string> cvar_client_destination(
+    "rcon.client.destination",
+    "Destination address for rcon commands, if empty the current server.",
+    Cvar::NONE,
+    ""
+);
+
+Message::Message( const netadr_t& remote, std::string command,
+				  Secure secure, std::string password, std::string challenge )
+	: secure_(secure),
+	  challenge_(std::move(challenge)),
+	  command_(std::move(command)),
+	  password_(std::move(password)),
+	  remote_(remote)
+{}
+
+Message::Message( std::string error_message )
+	: error_(std::move(error_message))
+{}
+
+const std::string& Message::command() const
+{
+	return command_;
```

Comment:
In general method names are CamelCase, we use GetCommand and don't postfix private members with _